### PR TITLE
Bulk sync queue and performance tweaks

### PR DESCRIPTION
**Pull request by skwp**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -16,6 +16,17 @@ The result of the sync is logged to a sync log available from the settings scree
...
+## Requirements
+
+In order for the cron to successfully process the listings sync in parallel-thread manner, the magento cron needs to be
+declared in the crontab via the following:
+
+php cron.php -mdefault 1
+
+or
+
+/bin/sh /var/www/html/reverb_magento/cron.sh cron.php -mdefault 1 > /dev/null 2>&1 &
```
Comment (skwp) [inline README.md]: not sure where this path is supposed to lead. where is cron.sh? @dunagan5887

File: README.md [hunk 2]
```diff
@@ -16,6 +16,17 @@ The result of the sync is logged to a sync log available from the settings scree
 
 Only simple products are synced. Configurable products are not synced.
 
+## Requirements
+
+In order for the cron to successfully process the listings sync in parallel-thread manner, the magento cron needs to be
+declared in the crontab via the following:
+
+php cron.php -mdefault 1
```
Comment (skwp) [inline README.md]: what is the full format of the line in crontab as it should appear? can we give a command to modify the crontab. something like this

```
(crontab -l ; echo "0 * * * * php /path/to/magento/app/reverb/cron.php") | crontab -
```
I made up the path as I'm not sure where that would live.

Comment (dunagan5887) [inline README.md]: Yan,

The lines above were given as they should appear.

php /path/to/magento/cron.php -mdefault 1


or


/bin/sh /path/to/magento/cron.sh cron.php -mdefault 1 > /dev/null 2>&1

are the actual lines which should appear in the crontab (only one of
them, not both of them)

Comment (skwp) [inline README.md]: But...this is not cron syntax? Am I missing something? Shouldn't every line start with timing information? "* * * * *"

Comment (dunagan5887) [inline README.md]: Ah, of course, I had taken for granted that the schedule syntax would be *
* * * , was focused more on the command aspect. I'll be sure to include
that in the README.md update

**Comment by skwp** [comment]
Thanks. When it comes to readmes, assume the reader has zero tech knowledge as a lot of our customers will be very basic on that front. The more self-installing we can make this the better. I'm actually thinking of an "install.sh" that would set up the cron for them.

File: README.md [hunk 3]
```diff
@@ -16,6 +16,24 @@ The result of the sync is logged to a sync log available from the settings scree
...
+
+In order for the cron to successfully process the listings sync in parallel-thread manner, the magento cron needs to be
+declared in the crontab via the following:
+
+* * * * * php /path/to/magento/cron.php -mdefault 1
+
+or
+
+* * * * *  /bin/sh /path/to/magento/cron.sh cron.php -mdefault 1 > /dev/null 2>&1 &
+
```
Comment (skwp) [inline README.md]: Is there any preferred way to do this? Is the magento one safer or more brittle across versions? Why would I use one vs the other?

**Comment by dunagan5887** [comment]
I have always used the php version, the .sh version just executes the "php cron.php" command, so I'm not sure why it would be used. As for the install.sh that sounds a bit complex; in the event that a user had

* * * * * php /path/to/magento/cron.php

the script would need to update this to be * * * * * php /path/to/magento/cron.php -mdefault 1, which is a bit more complex than just adding * * * * * php /path/to/magento/cron.php -mdefault 1 to the crontab. I would need to brush up on my Linux shell scripting for this, but it's something I could look into

**Comment by skwp** [comment]
To clarify, is the cron.php the global magento cron or is it pointing to our app's php script?

**Comment by skwp** [comment]
And what does "mdefault 1" do?

**Comment by dunagan5887** [comment]
cron.php is the global magento cron, it kicks off all crontab jobs listed in Magento. Given that there is no specific file to launch the Reverb cron job at this point, it's possible that another third-party extension's cron processes could continually error out and prevent the Reverb listing Sync from firing. In this event, an additional cron entry would be needed specifically for the Reverb listing Sync, and the listing sync crontab job would need to be removed from the ReverbSync module's config.xml file

mdefault 1 signifies Mode: Default, it's necessary to get threads to execute in parallel.

**Pull request merged**